tests/test_ERC20.py

```diff
@@ -1,41 +1,194 @@
+#############################################
+
+###############
+## approve() ##
+###############
+
+@pytest.mark.asyncio
+async def test_approve_from_caller():
+    _, erc20, owner, friend = await erc20_factory()
+    mint_amount = uint(1000)
+    approved_amount = uint(500)
+    # Mint tokens
+    await erc20.mint(owner.contract_address, mint_amount).invoke(caller_address=owner.contract_address)
+    # Then approve
+    await erc20.approve(friend.contract_address, approved_amount).invoke(caller_address=owner.contract_address)
+    # Check if the user is approved
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == approved_amount
+
+@pytest.mark.asyncio
+async def test_approve_none_minted():
+    _, erc20, owner, friend = await erc20_factory()
+    amount = uint(69420)
+    await erc20.approve(friend.contract_address, amount).invoke(owner.contract_address)
+    # Check if the user is approved
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == amount
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+
+@pytest.mark.asyncio
+async def test_fail_approve_zero_spender():
+    _, erc20, owner, _ = await erc20_factory()
+    amount = uint(69420)
+    await erc20.mint(owner.contract_address, amount).invoke(caller_address=owner.contract_address)
+    await assert_invoked_revert(erc20.approve(0, amount), owner.contract_address)
+
+##########################
+## increase_allowance() ##
+##########################
+
+@pytest.mark.asyncio
+async def test_increase_allowance():
+    _, erc20, owner, friend = await erc20_factory()
+    half = uint(500)
+    amount = uint(1000)
+    # Approve the user to spend the tokens
+    await erc20.approve(friend.contract_address, half).invoke(caller_address=owner.contract_address)
+    # Check if the user is approved to spend
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == half
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+    # Increase Allowance
+    executed = await erc20.increase_allowance(friend.contract_address, half).invoke(caller_address=owner.contract_address)
+    assert executed.result.success == 1
+    # Check increased allowance
+    executed_info = await erc20.allowance(owner.contract_address, friend.contract_address).call()
+    assert executed_info.result.allowance == amount
+    executed_info = await erc20.allowance(friend.contract_address, owner.contract_address).call()
+    assert executed_info.result.allowance == uint(0)
+
+@pytest.mark.asyncio
+async def test_increase_allowance_overflow():
```

Comment:
For more clarity regarding the expected behavior of the test, the function should have the `test_fail` prefix imho